<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>PerturbPlan App Documentation • perturbplanapp</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PerturbPlan App Documentation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">perturbplanapp</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/perturbplanapp.html">Documentation</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://katsevich-lab-perturbplan.share.connect.posit.cloud/"><span class="fa fa-external-link-alt"></span> Web App</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>PerturbPlan App Documentation</h1>
            
      

      <div class="d-none name"><code>perturbplanapp.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="design-problems">Design problems<a class="anchor" aria-label="anchor" href="#design-problems"></a>
</h2>
<div class="section level3">
<h3 id="steps-to-specify-a-design-problem">Steps to specify a design problem<a class="anchor" aria-label="anchor" href="#steps-to-specify-a-design-problem"></a>
</h3>
<div class="section level4">
<h4 id="assay-type">1. Assay type<a class="anchor" aria-label="anchor" href="#assay-type"></a>
</h4>
<p>The first step is to choose the assay type: Perturb-seq or TAP-seq.
PerturbPlan currently supports the same set of design problems for
Perturb-seq and TAP-seq. In particular, PerturbPlan does not currently
support TAP-seq-specific design problems, such as choosing which set of
genes to target. The choice of assay type primarily impacts the choice
of reference expression data and the quantification of gene expression
(see below).</p>
</div>
<div class="section level4">
<h4 id="optimization-constraints">2. Optimization constraints<a class="anchor" aria-label="anchor" href="#optimization-constraints"></a>
</h4>
<p>PerturbPlan facilitates constraining the power and cost of an
experiment.</p>
<ul>
<li><p><strong>Power.</strong> The target power for an experiment is
often set at 80% (the PerturbPlan default), which means that one expects
to detect at least 80% of the true perturbation-gene pairs. Power is
impacted by several choices; for example, power grows when profiling a
larger number of cells or sequencing a larger number of reads per
cell.</p></li>
<li><p><strong>Cost.</strong> The total cost is computed as the sum of
the library preparation cost and the sequencing cost. How these costs
are determined depends on the library preparation and sequencing
platforms used. To maintain flexibility across platforms, PerturbPlan
quantifies library preparation cost in terms of cost/cell and sequencing
cost in terms of cost/million reads. The defaults built into PerturbPlan
are based on the cost in 2025 of 10X Genomics v4 3’ library preparation
and Novaseq X 25B.</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="minimization-targets">3. Minimization targets<a class="anchor" aria-label="anchor" href="#minimization-targets"></a>
</h4>
<p>Given power and/or cost constraints, PerturbPlan supports the
minimization of five parameters:</p>
<ul>
<li><p><strong>Cells per target</strong>: The average number of cells
receiving CRISPR perturbations of a given target. Increasing the number
of cells per target increases power, but also increases cost. Minimizing
this parameter can be used as a proxy for minimizing cost (often while
constraining power).</p></li>
<li><p><strong>Reads per cell</strong>: The average number of sequenced
reads per cell. Increasing the number of reads per cell increases power,
but also increases cost. Minimizing this parameter can be used as a
proxy for minimizing cost (often while constraining power).</p></li>
<li><p><strong>Expression threshold</strong>: Even if the
experimentalist has in mind a concrete set of perturbation-gene pairs to
test for association, pairs involving lowly expressed genes can drag
down the power of the analysis. For this reason, a minimum gene
expression threshold can be applied to filter the set of pairs prior to
testing. Minimizing this threshold is a means to include as many genes
as possible in the analysis, while still achieving the target
power.</p></li>
<li><p><strong>Fold change</strong>: The fold change measures the size
of the effect of a perturbation on a gene’s expression (see below for a
more precise definition). Minimizing the fold change is a means to be
able to detect more subtle effects, while still achieving the target
power.</p></li>
<li><p><strong>Total cost</strong>: Note that cost can be a constraint
(as above) or a variable to be minimized over experimental parameters,
subject to a power constraint.</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="other-varying-parameters">4. Other varying parameters<a class="anchor" aria-label="anchor" href="#other-varying-parameters"></a>
</h4>
<p>In some of the design problems, parameters in addition to that being
minimized are allowed to vary. These include cells/target and
reads/cell; see below for more detail.</p>
</div>
</div>
<div class="section level3">
<h3 id="supported-design-problems">Supported design problems<a class="anchor" aria-label="anchor" href="#supported-design-problems"></a>
</h3>
<p>PerturbPlan requires a power constraint for all design problems, and
allows choosing whether to include a cost constraint as well. All
supported combinations for optimization constraints, minimization
targets, and other varying parameters are presented in the table
below.</p>
<table class="table">
<colgroup>
<col width="2%">
<col width="23%">
<col width="33%">
<col width="40%">
</colgroup>
<thead><tr class="header">
<th></th>
<th><strong>Constraints</strong></th>
<th><strong>Minimization target</strong></th>
<th><strong>Other varying parameters</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Power</td>
<td>Cells/Target</td>
<td>–</td>
</tr>
<tr class="even">
<td>2</td>
<td>Power</td>
<td>Reads/Cell</td>
<td>–</td>
</tr>
<tr class="odd">
<td>3</td>
<td>Power</td>
<td>Expression threshold</td>
<td>–</td>
</tr>
<tr class="even">
<td>4</td>
<td>Power</td>
<td>Fold change</td>
<td>–</td>
</tr>
<tr class="odd">
<td>5</td>
<td>Power</td>
<td>Total cost</td>
<td>Cells/Target + Reads/Cell</td>
</tr>
<tr class="even">
<td>6</td>
<td>Power + Cost</td>
<td>Expression threshold</td>
<td>Cells/Target</td>
</tr>
<tr class="odd">
<td>7</td>
<td>Power + Cost</td>
<td>Expression threshold</td>
<td>Reads/Cell</td>
</tr>
<tr class="even">
<td>8</td>
<td>Power + Cost</td>
<td>Expression threshold</td>
<td>Cells/Target + Reads/Cell</td>
</tr>
<tr class="odd">
<td>9</td>
<td>Power + Cost</td>
<td>Fold change</td>
<td>Cells/Target</td>
</tr>
<tr class="even">
<td>10</td>
<td>Power + Cost</td>
<td>Fold change</td>
<td>Reads/Cell</td>
</tr>
<tr class="odd">
<td>11</td>
<td>Power + Cost</td>
<td>Fold change</td>
<td>Cells/Target + Reads/Cell</td>
</tr>
</tbody>
</table>
<p>Design problems 1-4 are the simplest: they determine the minimum
value of one parameter required to attain the desired power, while
holding all other parameters fixed. Design problem 5 allows cells/target
and reads/cell to vary, finding the lowest-cost combination to achieve a
target power. The remainder of the design problems involve constraining
both cost and power, allowing the minimization of either expression
threshold or fold change, while allowing one or both of cells/target and
reads/cell to vary.</p>
</div>
</div>
<div class="section level2">
<h2 id="design-parameters">Design parameters<a class="anchor" aria-label="anchor" href="#design-parameters"></a>
</h2>
<p>Once you have selected your design problem, you need to specify the
design parameters that define your screen. These parameters are grouped
into four collapsible sections: <strong>experimental choices</strong>
(how the data will be collected), <strong>analysis choices</strong> (how
the data will be analyzed), <strong>effect sizes</strong> (the strength
of the biological effects you wish you detect), and <strong>advanced
settings</strong> (optional additional settings).</p>
<hr>
<div class="section level4">
<h4 id="experimental-choices">Experimental choices<a class="anchor" aria-label="anchor" href="#experimental-choices"></a>
</h4>
<div class="section level5">
<h5 id="reference-expression-data">Reference expression data<a class="anchor" aria-label="anchor" href="#reference-expression-data"></a>
</h5>
<p>Reference expression data for the biological system of interest is
needed to compute power. These data include baseline mean expressions
for all expressed genes (Perturb-seq) or targeted genes (TAP-seq), a
fitted sequencing saturation curve, and a mapping efficiency (defined
below). Reference expression data can be either built-in or custom:</p>
<ul>
<li><p><strong>Built-in</strong>: PerturbPlan ships with reference
expression data from six cell types commonly used in perturbation
screens (K562, A549, THP1, CD8+ T, iPSC, iPSC-derived neurons),
documented <a href="https://katsevich-lab.github.io/perturbplan/reference/index.html#example-data" class="external-link">here</a>.
Consider using these if you are planning a Perturb-seq experiment and
your cell type falls into one of these categories.</p></li>
<li><p><strong>Custom</strong>: Custom reference data can be uploaded to
the PerturbPlan app. The required format and the means to process your
data into this format are documented <a href="https://katsevich-lab.github.io/perturbplan/reference/reference_data_processing.html#value" class="external-link">here</a>
and <a href="https://katsevich-lab.github.io/perturbplan/articles/preprocess-reference.html" class="external-link">here</a>.
Use custom reference data if you are planning a Perturb-seq experiment
in a cell type not covered by the built-in data, or if you are planning
a TAP-seq experiment. Since each TAP-seq experiment comes with a unique
set of targeted genes, PerturbPlan does not include built-in TAP-seq
reference data. If you wish to try out the TAP-seq functionality of the
app, you may <a href="https://katsevich-lab.github.io/perturbplan/reference/K562_Ray.html" class="external-link">download</a>
a sample reference expression dataset, documented <a href="https://katsevich-lab.github.io/perturbplan/reference/K562_Ray.html" class="external-link">here</a>.</p></li>
</ul>
</div>
<div class="section level5">
<h5 id="multiplicity-of-infection-moi">Multiplicity of infection (MOI)<a class="anchor" aria-label="anchor" href="#multiplicity-of-infection-moi"></a>
</h5>
<p>The average number of gRNA-carrying lentiviral particles that infect
a single cell.</p>
</div>
<div class="section level5">
<h5 id="number-of-perturbation-targets">Number of perturbation targets<a class="anchor" aria-label="anchor" href="#number-of-perturbation-targets"></a>
</h5>
<p>The number of genomic elements (commonly, genes or enhancers) you
plan to perturb in your screen. Not to be confused with the number of
gene targeted for sequencing in a TAP-seq experiment.</p>
</div>
<div class="section level5">
<h5 id="grnas-per-target">gRNAs per target<a class="anchor" aria-label="anchor" href="#grnas-per-target"></a>
</h5>
<p>The number of gRNAs designed to perturb each target. Having multiple
gRNAs per target improves robustness against ineffective guides and
improves power by averaging noisy gRNA effects.</p>
</div>
<div class="section level5">
<h5 id="number-of-non-targeting-grnas">Number of non-targeting gRNAs<a class="anchor" aria-label="anchor" href="#number-of-non-targeting-grnas"></a>
</h5>
<p>Non-targeting gRNAs are control gRNAs that do not target any genomic
element. They are used to form a control group of cells (in low-MOI
screens) and to check for calibration (in all screens). While the effect
of the number of non-targeting gRNAs on power is usually small, their
utility is for other purposes. It is recommended to include tens or even
hundreds of non-targeting gRNAs in perturbation screens.</p>
</div>
<div class="section level5">
<h5 id="cells-per-target">Cells per target<a class="anchor" aria-label="anchor" href="#cells-per-target"></a>
</h5>
<p>The average number of cells receiving CRISPR perturbations of a given
target, equal to the total number of cells profiled divided by the
number of perturbation targets.</p>
</div>
<div class="section level5">
<h5 id="reads-per-cell">Reads per cell:<a class="anchor" aria-label="anchor" href="#reads-per-cell"></a>
</h5>
<p>The average number of sequenced reads per cell. More sequenced reads
gives rise to more unique molecular identifiers (UMIs) per cell, which
leads to less noisy gene expressions and therefore higher power.</p>
</div>
</div>
<div class="section level4">
<h4 id="analysis-choices">Analysis Choices<a class="anchor" aria-label="anchor" href="#analysis-choices"></a>
</h4>
<div class="section level5">
<h5 id="perturbation-gene-pairs-to-analyze">Perturbation-gene pairs to analyze<a class="anchor" aria-label="anchor" href="#perturbation-gene-pairs-to-analyze"></a>
</h5>
<p>The set of perturbation-gene pairs you plan to test for association.
These pairs can be specified in one of two ways:</p>
<ul>
<li><p><strong>Random pairing</strong>: If you have not yet decided
which perturbation-gene pairs to analyze, you can choose to pair
perturbations to genes at random, subject to the constraint that the
genes involved survive the expression threshold.</p></li>
<li><p><strong>Custom pairing</strong>: If you have decided which
perturbation-gene pairs to analyze, you can upload this set of pairs to
PerturbPlan as an RDS file containing a data frame with two columns
named <code>grna_target</code> (an identifier of the genomic element
being targeted) and <code>response_id</code> (the ENSG ID of the
gene).</p></li>
</ul>
</div>
<div class="section level5">
<h5 id="test-sidedness">Test sidedness<a class="anchor" aria-label="anchor" href="#test-sidedness"></a>
</h5>
<p>The expected direction of the perturbation effect. The options
are</p>
<ul>
<li><p><strong>Left</strong>: If perturbing the genomic element is
expected to down-regulate the gene it is tested against.</p></li>
<li><p><strong>Right</strong>: If perturbing the genomic element is
expected to up-regulate the gene it is tested against.</p></li>
<li><p><strong>Both</strong>: If the direction of the effect is
unknown.</p></li>
</ul>
</div>
<div class="section level5">
<h5 id="expression-threshold">Expression threshold<a class="anchor" aria-label="anchor" href="#expression-threshold"></a>
</h5>
<p>The relevant quantification of gene expression depends on whether a
Perturb-seq or TAP-seq is being planned:</p>
<ul>
<li><p><strong>Perturb-seq</strong>: Expressions obtained from
transcriptome-wide RNA-seq are commonly measured in the units of
<strong>transcripts per million (TPM)</strong>, the expected number of
UMIs coming from a given gene among a million UMIs captured in a
cell.</p></li>
<li><p><strong>TAP-seq</strong>: Measuring expression in terms of TPMs
is less meaningful in a TAP-seq experiment, where expression is measured
for a smaller number of genes. As an interpretable substitute for TPM,
we quantify TAP-seq gene expression in terms of the <strong>UMIs/cell at
saturation</strong>, the expected number of UMIs coming from a given
gene in a cell if sequenced to saturation.</p></li>
</ul>
</div>
</div>
<div class="section level4">
<h4 id="effect-sizes">Effect sizes<a class="anchor" aria-label="anchor" href="#effect-sizes"></a>
</h4>
<p>The following two parameters are relevant for defining the biological
effects you wish to detect.</p>
<div class="section level5">
<h5 id="fold-change">Fold change<a class="anchor" aria-label="anchor" href="#fold-change"></a>
</h5>
<p>The fold change is the multiplicative change in mean gene expression
induced by the perturbation:</p>
<ul>
<li><p><strong>Fold change = 1.0</strong>: No effect (null
hypothesis).</p></li>
<li><p><strong>Fold change &lt; 1.0</strong>: Negative/inhibitory effect
(e.g., 0.8 = 20% decrease in expression).</p></li>
<li><p><strong>Fold change &gt; 1.0</strong>: Positive/activating effect
(e.g., 1.5 = 50% increase in expression).</p></li>
</ul>
<p>For a conservative power estimate, specify the <strong>weakest effect
size of interest.</strong></p>
</div>
<div class="section level5">
<h5 id="proportion-of-non-null-pairs">Proportion of non-null pairs<a class="anchor" aria-label="anchor" href="#proportion-of-non-null-pairs"></a>
</h5>
<p>The fraction of tested perturbation-gene pairs expected to have
signal at least a strong as the weakest effect size of interest. Higher
proportions of non-null pairs increase power by reducing the impact of
multiple testing corrections. Estimating this parameter for perturbation
screens remains an open question, but the default value of 0.01 is a
reasonable starting point.</p>
</div>
</div>
<div class="section level4">
<h4 id="advanced-settings">Advanced settings<a class="anchor" aria-label="anchor" href="#advanced-settings"></a>
</h4>
<div class="section level5">
<h5 id="grna-variability">gRNA variability<a class="anchor" aria-label="anchor" href="#grna-variability"></a>
</h5>
<p>The variance of the per-gRNA fold changes for each target. If the
fold change for a given perturbation-gene pair is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>β</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>
and the gRNA variability is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>σ</mi><mn>2</mn></msup><annotation encoding="application/x-tex">\sigma^2</annotation></semantics></math>,
then the fold change induced by a gRNA is modeled as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>N</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>β</mi><mo>,</mo><msup><mi>σ</mi><mn>2</mn></msup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">N(\beta, \sigma^2)</annotation></semantics></math>.</p>
</div>
<div class="section level5">
<h5 id="mapping-efficiency">Mapping efficiency<a class="anchor" aria-label="anchor" href="#mapping-efficiency"></a>
</h5>
<p>The fraction of sequencing reads that map confidently to the genes of
interest. This quantity is derived from the reference expression data,
but may be changed manually if desired. The mapping efficiency depends
on the type of experiment being planned:</p>
<ul>
<li><p><strong>Perturb-seq</strong>: In Perturb-seq, all genes are of
interest, so mapping efficiency is the proportion of reads mapping to
any gene. Perturb-seq mapping efficiencies often fall in the range
0.65-0.75.</p></li>
<li><p><strong>TAP-seq</strong>: Imperfect gene-specific primers can
hybridize to transcripts coming from genes other than those targeted.
Therefore, a proportion of reads confidently mapped to the genome do not
map to the target genes, making TAP-seq mapping efficiencies generally
lower than those of Perturb-seq. For example, the mapping efficiency of
the recent Ray et al. (2025) TAP-seq dataset was 0.35.</p></li>
</ul>
</div>
<div class="section level5">
<h5 id="control-group">Control group<a class="anchor" aria-label="anchor" href="#control-group"></a>
</h5>
<p>The strategy used to construct control cells for testing perturbation
effects.</p>
<ul>
<li><p><strong>Non-targeting cells</strong>: Cells that receive only
non-targeting gRNAs. This is the default choice for low-MOI screens (MOI
&gt; 1).</p></li>
<li><p><strong>Complement cells</strong>: Cells that do not receive the
perturbation being tested, but may receive other perturbations. This is
the required choice for high-MOI experiments (MOI &gt; 1), since few
cells will have exclusively non-targeting gRNAs.</p></li>
</ul>
<p>For more on this choice, see the <a href="https://timothy-barry.github.io/sceptre-book/set-analysis-parameters.html#sec-set-analysis-parameters_control_group" class="external-link">sceptre
documentation</a>.</p>
</div>
<div class="section level5">
<h5 id="fdr-target-level">FDR target level<a class="anchor" aria-label="anchor" href="#fdr-target-level"></a>
</h5>
<p>The maximum tolerated expected proportion of discovered associations
that are false positives. The FDR target level of 0.1 is the most common
choice.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ziang Niu, Yihui He, Eugene Katsevich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
